hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala

```diff
@@ -146,6 +172,274 @@ object HLESurveyTransformationPipelineBuilder extends PipelineBuilder[Args] {
       ocSecondaryResidenceZip = None,
       ocSecondaryResidenceOwnership = None,
       ocSecondaryResidenceOwnershipOther = None,
-      ocSecondaryResidenceTimePercentage = None
+      ocSecondaryResidenceTimePercentage = None,
+      ddOtherRecentShortTermResidences = Array.empty
+    )
+
+    val withBreed = {
+      val isPureBreed = rawRecord.getRequired("dd_dog_pure_or_mixed") == "1"
+      if (isPureBreed) {
+        val breed = rawRecord.getOptional("dd_dog_breed")
+        dogBase.copy(
+          ddBreedPure = breed,
+          ddBreedPureNonAkc =
+            if (breed.contains("277")) rawRecord.getOptional("dd_dog_breed_non_akc") else None
+        )
+      } else {
+        dogBase.copy(
+          ddBreedMixedPrimary = rawRecord.getOptional("dd_dog_breed_mix_1"),
+          ddBreedMixedSecondary = rawRecord.getOptional("dd_dog_breed_mix_2")
+        )
+      }
+    }
+
+    val withAge = {
+      val ageCertain = rawRecord.getRequired("dd_dog_birth_year_certain") == "1"
+
+      /*
+       * FIXME: How do we encode this in the data?
+       *
+       * Age basis codebook:
+       *  - '1' == Calculated from birth year and month
+       *  - '2' == Estimated from birth year
+       *  - '3' == Estimated by owner
+       */
+
+      if (ageCertain) {
+        val formYear = rawRecord.getRequired("dd_dog_current_year_calc").toInt
+        val birthYear = rawRecord.getRequired("dd_dog_birth_year").toInt
+
+        val formMonth = rawRecord.getRequired("dd_dog_current_month_calc").toInt
+        val exactMonthKnown = rawRecord.getRequired("dd_dog_birth_month_yn") == "1"
+        val birthMonth = if (exactMonthKnown) {
+          rawRecord.getRequired("dd_dog_birth_month").toInt
+        } else {
+          // ¯\_(ツ)_/¯
+          6
```

Comment:
lol is this temporary?

Reply:
love it lol

Reply:
I think it accurately captures my feelings of using `6` here. I'll add some words.